evm/src/modules/TokenGateway.sol

```diff
@@ -35,6 +43,13 @@ struct TeleportParams {
     // calculated amountInMax: 
     // used if selected fee token is not expected fee token
     uint256 amountInMax;
+    // Permit Details & Signature for host to spend feeToken
+    TeleportPermit hostFeeTokenPermit;
+    // Permit Details & Signature for gateway to spend asset
+    TeleportPermit gatewayAssetPermit;
+    // Permit Details & Signature for gateway to user selected fee token
+    // This is useful when the selected fee token is not the required fee token
+    TeleportPermit gatewayInputFeeTokenPermit;
```

Comment:
uhm i think if this is the case, might be better to just do a batch permit?

https://docs.uniswap.org/contracts/permit2/reference/allowance-transfer#batched-permit